tests/ga/test_extension.py

```diff
@@ -387,6 +390,27 @@ def _set_up_update_test_and_update_gs(self, patch_command, *args):
 
         return test_data, exthandlers_handler, protocol
 
+    def _setup_extension_for_validating_collect_ext_status(self, mock_lib_dir, status_file, *args):
+        handler_name = "TestHandler"
+        handler_version = "1.0.0"
+        mock_lib_dir.return_value = self.lib_dir
+        fileutil.mkdir(os.path.join(self.lib_dir, handler_name + "-" + handler_version, "config"))
+        fileutil.mkdir(os.path.join(self.lib_dir, handler_name + "-" + handler_version, "status"))
+        shutil.copy(os.path.join(data_dir, "ext", status_file),
+                    os.path.join(self.lib_dir, handler_name + "-" + handler_version, "status", "0.status"))
+        shutil.copy(tempfile.mkstemp(prefix="test-file")[1],
+                    os.path.join(self.lib_dir, handler_name + "-" + handler_version, "config", "0.settings"))
+
+        test_data = mockwiredata.WireProtocolData(mockwiredata.DATA_FILE)
```

Comment:
for new tests let's use mock_wire_protocol instead of mockwiredata.

the problem with using mockwiredata is that each test needs to create their mocks for http get and crypt utils. When there is need to change something on the mock (e.g. when i had to add support for POST or mock http request instead of http get) the impact on tests can be huge.

Reply:
btw, as i continue refactoring protocol very likely i will have to change those mocks again... i am not looking forward to all the tests update i will have to do